src/Build/BackEnd/Components/Logging/LoggingService.cs

```diff
@@ -404,96 +404,81 @@ public int NextTaskId
         {
-            get
-            {
-                return _onlyLogCriticalEvents;
-            }
+            get => _onlyLogCriticalEvents;
 
-            set
-            {
-                _onlyLogCriticalEvents = value;
-            }
+            set => _onlyLogCriticalEvents = value;
         }
 
         /// <summary>
         /// Number of nodes in the system when the system is initially started
         /// </summary>
         public int MaxCPUCount
         {
-            get
-            {
-                return _maxCPUCount;
-            }
+            get => _maxCPUCount;
 
-            set
-            {
-                _maxCPUCount = value;
-            }
+            set => _maxCPUCount = value;
         }
 
         /// <summary>
         /// The list of descriptions which describe how to create forwarding loggers on a node.
         /// This is used by the node provider to get a list of registered descriptions so that 
         /// they can be transmitted to child nodes.
         /// </summary>
-        public ICollection<LoggerDescription> LoggerDescriptions
-        {
-            get
-            {
-                return _loggerDescriptions;
-            }
-        }
+        public ICollection<LoggerDescription> LoggerDescriptions => _loggerDescriptions;
 
         /// <summary>
         /// Enumerator over all registered loggers.
         /// </summary>
-        public ICollection<ILogger> Loggers
-        {
-            get { return _iloggerList; }
-        }
+        public ICollection<ILogger> Loggers => _loggers;
 
         /// <summary>
         /// What type of logging mode is the logger running under. 
         /// Is it Synchronous or Asynchronous
         /// </summary>
-        public LoggerMode LoggingMode
-        {
-            get
-            {
-                return _logMode;
-            }
-        }
+        public LoggerMode LoggingMode => _logMode;
 
         /// <summary>
         /// Get of warnings to treat as errors.  An empty non-null set will treat all warnings as errors.
         /// </summary>
         public ISet<string> WarningsAsErrors
         {
-            get { return _warningsAsErrors; }
-            set { _warningsAsErrors = value; }
-        }
+            get;
+            set;
+        } = null;
 
         /// <summary>
         /// A list of warnings to treat as low importance messages.
         /// </summary>
         public ISet<string> WarningsAsMessages
         {
-            get { return _warningsAsMessages; }
-            set { _warningsAsMessages = value; }
+            get;
+            set;
+        } = null;
+
+        /// <summary>
+        /// Should evaluation events include profiling information?
+        /// </summary>
+        public bool IncludeEvaluationProfile
+        {
+            get => _includeEvaluationProfile || _eventSinkDictionary.Values.OfType<EventSourceSink>().Any(sink => sink.IncludeEvaluationProfiles);
```

Comment:
These linqs will execute on each evaluation start, which might get picked up in small project restore perf tests (.net core perf is pretty critical about this scenario). Could you hide them behind a Lazy?